massa-ledger-worker/src/ledger_db.rs

```diff
@@ -391,17 +403,21 @@ impl LedgerDB {
         &self,
         handle: &ColumnFamily,
         batch: &mut LedgerBatch,
-        key: &[u8],
+        key: &Key,
         value: &[u8],
     ) {
+        let mut serialized_key = Vec::new();
+        self.key_serializer
+            .serialize(key, &mut serialized_key)
+            .expect(KEY_SER_ERROR);
         let mut len_bytes = Vec::new();
         self.len_serializer
-            .serialize(&(key.len() as u64), &mut len_bytes)
+            .serialize(&(serialized_key.len() as u64), &mut len_bytes)
             .expect(KEY_LEN_SER_ERROR);
-        let hash = Hash::compute_from(&[&len_bytes, key, value].concat());
+        let hash = Hash::compute_from(&[&len_bytes, &serialized_key, value].concat());
         batch.ledger_hash ^= hash;
-        batch.aeh_list.insert(key.to_vec(), hash);
-        batch.write_batch.put_cf(handle, key, value);
+        batch.aeh_list.insert(serialized_key.to_vec(), hash);
```

Comment:
To vec here shouldn't be necessary as it's already a vec. if it's to clone it because it's moved the line above you can use the `clone()` combinator : 
```suggestion
        batch.aeh_list.insert(serialized_key.clone(), hash);
```